test/quantization/test_quantized_op.py

```diff
@@ -1838,6 +1838,25 @@ def test_mean(self, X, dim):
 
         self.assertEqual(Y, qY.dequantize())
 
+    @given(engine=st.one_of(st.just('fbgemm'), st.just('qnnpack')), keep=st.booleans())
+    @override_qengines
+    def test_quantized_mean(self, engine, keep):
+        if engine not in torch.backends.quantized.supported_engines:
+            skipTest("This Pytorch Build has not been built with or does not support {0}".format(engine.toupper()))
+        else:
+            torch.backends.quantized.engine = engine
```

Comment:
There is `with override_quantized_engine('qnnpack'):` context manager that we should use here. e.g., https://github.com/pytorch/pytorch/blob/master/test/quantization/core/test_quantized_op.py#L434.

That way test environment is not polluted. If default engine was fbgemm then we keep it same with context manager and only run the test with different engine.

Reply:
what would happen if I was using `with override_quantized_engine('qnnpack')` on the box that doesn't support QNNPACK?